File: fix_script.py
import re

filepath = "/Users/a12345678/Pliki/Programowanie/neuroshit/script.js"
with open(filepath, "r", encoding="utf8") as f:
    text = f.read()

# Pass 'importance' attribute to allEdges
text = text.replace(
'''      allEdges.push({
        from: conn.from,
        to: conn.to,
        connectionType: conn.connectionType,
        system: system.id,
      });''',
'''      allEdges.push({
        from: conn.from,
        to: conn.to,
        connectionType: conn.connectionType,
        system: system.id,
        importance: conn.importance || "main",
      });'''
)

# And in buildSynthesisChains
text = text.replace(
'''            synthesisChains.push({
              substrate, enzyme, product,
              chainType,
              system: inEdge.system,
            });''',
'''            synthesisChains.push({
              substrate, enzyme, product,
              chainType,
              system: inEdge.system,
              importance: outEdge.importance || "main",
            });'''
)

# Render them dim if secondary
text = re.sub(
r'(function drawEdge\(edge\) \{\n.*const style = \{ \.\.\.CONNECTION_TYPES\[edge\.connectionType\] \};.*?)\n',
r'\1\n    if (edge.importance === "secondary") {\n      style.color = style.color.replace(/[\d\.]+\)$/, "0.15)");\n      ctx.globalAlpha = 0.4;\n    }\n',
text, count=1, flags=re.DOTALL|re.MULTILINE
)

# And we must restore globalAlpha at end of drawEdge
text = re.sub(
r'(    ctx.fill\(\);\n  \})',
r'    ctx.fill();\n    ctx.globalAlpha = 1.0;\n  }',
text
)

# Also in drawSynthesisChain
text = re.sub(
r'(function drawSynthesisChain\(chain\) \{\n.*?const style = \{ \.\.\.(?:.*?CONNECTION_TYPES\.(?:synthesis|degradation)) \};.*?)\n',
r'\1\n    if (chain.importance === "secondary") {\n      style.color = style.color.replace(/[\d\.]+\)$/, "0.15)");\n      ctx.globalAlpha = 0.4;\n    }\n',
text, count=1, flags=re.DOTALL|re.MULTILINE
)

# Wait, the spreading of CONNECTION_TYPES might not be happening in drawSynthesisChain, let's check it.
with open(filepath, "w", encoding="utf8") as f:
    f.write(text)
print("Done first step")

File: fix_script.js
const fs = require("fs");
const filepath = "/Users/a12345678/Pliki/Programowanie/neuroshit/script.js";
let text = fs.readFileSync(filepath, "utf8");

text = text.replace(
`      allEdges.push({
        from: conn.from,
        to: conn.to,
        connectionType: conn.connectionType,
        system: system.id,
      });`,
`      allEdges.push({
        from: conn.from,
        to: conn.to,
        connectionType: conn.connectionType,
        system: system.id,
        importance: conn.importance || "main",
      });`
);

text = text.replace(
`            synthesisChains.push({
              substrate, enzyme, product,
              chainType,
              system: inEdge.system,
            });`,
`            synthesisChains.push({
              substrate, enzyme, product,
              chainType,
              system: inEdge.system,
              importance: outEdge.importance || "main",
            });`
);

// Update drawEdge
text = text.replace(
  /const style = \{\s*\.\.\.CONNECTION_TYPES\[edge\.connectionType\]\s*\};/,
  `const style = { ...CONNECTION_TYPES[edge.connectionType] };
    if (edge.importance === "secondary") {
      style.color = style.color.replace(/rgba\\((.*?),\\s*[.\\d]+\\)/, "rgba($1, 0.15)");
    }`
);

// Update drawSynthesisChain
text = text.replace(
  /const style = chainType === "synthesis"[\s\S]*?\? CONNECTION_TYPES\.synthesis[\s\S]*?: CONNECTION_TYPES\.degradation;/,
  `const baseStyle = chainType === "synthesis"
      ? CONNECTION_TYPES.synthesis
      : CONNECTION_TYPES.degradation;
    const style = { ...baseStyle };
    if (chain.importance === "secondary") {
      style.color = style.color.replace(/rgba\\((.*?),\\s*[.\\d]+\\)/, "rgba($1, 0.15)");
    }`
);

fs.writeFileSync(filepath, text);
console.log("Done");
